test/functional/hosts_controller_test.rb

```diff
@@ -578,6 +578,10 @@ def test_unset_manage
     Setting[:restrict_registered_smart_proxies] = true
     Setting[:require_ssl_smart_proxies] = true
     SETTINGS[:require_ssl] = true
+    # Adding https and certificates or else the request will be redirected
+    @request.env['HTTPS'] = 'on'
```

Comment:
That means you're no longer testing the scenario described on line 576?  Instead it seems to test the SSL verification step over HTTPS.

That probably points to this PR needing to add something to disable the redirect in the Foreman::Controller::SmartProxyAuth concern, which currently disables the existing SSL redirect and checking require_ssl itself.